new-cep.md

```diff
@@ -0,0 +1,218 @@
+decorated function called either `conda_pre_command` or `conda_post_command` which return either a `CondaPreCommand` 
+or `CondaPostCommand` class, respectively. These classes will both receive the following three properties:
+
+- `name`: unique name which identifies this plugin hook
+- `action`: a callable which contains the code to be run
+- `run_for`: a Python `set` of strings representing the commands this will be run on (e.g. `install` and `create`)
+
+#### Example
+
+```python
+from conda.plugins import hookimpl, CondaPreCommand, CondaPostCommand
+
+
+PLUGIN_NAME = "custom_plugin"
+
+
+def custom_plugin_pre_command_action(command):
+    """
+    Defines our custom pre-run action which simply prints a message.
+    """
+    print(f"pre-run action before {command}")
+
+
+@hookimpl
+def conda_pre_commands():
+    """
+    Yields our CondaPreCommand instance which attaches our ``custom_plugin_pre_command_action``
+    to the "install" and "create" command.
+    """
+    yield CondaPreCommand(
+        name=f"{PLUGIN_NAME}_pre_command",
+        action=custom_plugin_pre_command_action,
+        run_for={"install", "create"}
+    )
+
+
+def custom_plugin_post_command_action(command):
+    """
+    Defines our custom post-run action which simply prints a message.
+    """
+    print(f"post-run action after {command}")
+
+
+@hookimpl
+def conda_post_commands():
+    """
+    Yields our CondaPreCommand instance which attaches our ``custom_plugin_post_command_action`` to
+    the "install" and "create" command.
+    """
+    yield CondaPostCommand(
+        name=f"{PLUGIN_NAME}_post_command",
+        action=custom_plugin_post_command_action,
+        run_for={"install", "create"}
+    )
+```
+
+### `exception_handler`
+
+For the `exception_handler` hook, plugin authors begin by defining a `conda.plugins.hookimpl` 
+decorated  function called `conda_exception_handler`. This function will return the
+ `CondaExceptionHandler` class with the following attributes:
+
+- `name`: unique name which identifies this plugin hook
+- `handler`: a class that is a subclass of [`conda.exceptions.ExceptionHandler`]
+
+#### Example
+
+```python
+from conda.exceptions import ExceptionHandler
+from conda.plugins import hookimpl, CondaExceptionHandler
+
+PLUGIN_NAME = "custom_plugin"
+
+
+class CustomExceptionHandler(ExceptionHandler):
+    """Custom implementation of the ``conda.exceptions.ExceptionHandler``"""
+
+    def handle_exception(self, exc_val, exc_tb):
+        print(
+            "Here's a custom message that will show when" 
+            " an exception is encountered!"
+        )
+
+        super().handle_exception(exc_val, exc_tb)
+
+
+@hookimpl
+def conda_on_exception():
+    """
+    Returns our ``CondaExceptionHandler`` class which attaches our 
+    registers our ``
+    """
+    yield CondaExceptionHandler(
+        name=f"{PLUGIN_NAME}_exception_handler",
+        handler=CustomExceptionHandler
```

Comment:
will it support `run_for` as well?

Reply:
I left that out because I didn't think that this plugin should have that fine grain control. It would also complicate the implementation too.